documentation/release_notes.markdown

```diff
@@ -61,179 +94,253 @@ This release retires all api versions before v4 and promotes v4 to be
 the stable API. The following changes have been made since the last
 stable release of PuppetDB, which included an experimental v4 API.
 
-* Retire all previously deprecated commands ([PDB-840](https://tickets.puppetlabs.com/browse/PDB-840))
+* Versions 2 and 3 of the query API are now retired. There are also significant differences between the current stable v4 and the previously experimental v4.
+
+    Full documentation outlining this change is available [in the API upgrading guide][upgrading].
+
+* We've retired some previously deprecated commands ([PDB-840](https://tickets.puppetlabs.com/browse/PDB-840))
 
     The currently supported commands and versions are now:
 
     * replace catalog, v6
     * replace facts, v4
-    * deactivate node, v2
+    * deactivate node, v3
     * store report, v5
 
-* All API commands and responses now use underscores, instead of dashes.  ([PDB-698](https://tickets.puppetlabs.com/browse/PDB-698))
+    Older versions of the commands on your queue will still be
+    processed by PuppetDB so you won't lose the data from messages
+    still being processed by PuppetDB before an upgrade.
+
+* All API commands and responses now use underscores, instead of dashes. ([PDB-698](https://tickets.puppetlabs.com/browse/PDB-698))
 
     This patch changes our API commands and responses to use underscores instead of
     dashes, which will make it easier to interface with PDB using languages like
-    javascript and python.
-
-* Split the `/v3/commands` endpoint into its own API served at `/pdb/cmd/v1`.
+    javascript and python. Consult the various API endpoints for the exact new
```

Comment:
"commands" is overloaded, and inaccurate in the context of space-separated PDB commands.

This patch changes our query fields and parameters to use underscores instead of dashes. Consult the documentation on the various API endpoints for specific information.